src/components/Thumbs.js

```diff
@@ -26,81 +26,85 @@ class Thumbs extends Component {
         super(props);
 
         this.state = {
-            initialized: false,
             selectedItem: props.selectedItem,
             hasMount: false,
-            firstItem: this.getFirstItem(props.selectedItem),
-            images: []
+            firstItem: 0,
+            itemSize: null,
+            visibleItems: 0,
+            lastPosition: 0,
+            showArrows: false,
+            images: this.getImages()
```

Comment:
Everything state is in `state` now so the relevant setState calls ensure we properly render immediately rather than waiting for a side-effect somewhere else to do it, and causing flickering etc.